core/build.gradle

```diff
@@ -44,12 +45,29 @@ android {
     sourceSets {
         androidTest.assets.srcDirs += files("$projectDir/schemas".toString())
     }
+
+    ndkVersion rootProject.ndkVersion
 }
 
 androidExtensions {
     experimental = true
 }
 
+cargo {
+    module  = 'src/main/rust'
+    libname = 'sslocal'
+    targets = ['arm', 'arm64', 'x86', 'x86_64']
+    profile = 'debug'
+    targetIncludes = ['libsslocal.so']
+    extraCargoBuildArguments = ['--bin', 'sslocal']
+    features {
+        noDefaultBut "sodium", "android", "rc4", "aes-cfb", "aes-ctr", "camellia-cfb", "openssl-vendored"
+    }
+    exec { spec, toolchain ->
+        spec.environment("RUSTFLAGS", "-C link-arg=-o -C link-arg=target/${toolchain.target}/${profile}/libsslocal.so")
```

Comment:
I would think that the plugin would take care of the output so name... Is this an upstream issue?

Reply:
I think it's actually a limitation of Cargo, which cannot customize the output binary name.

Reply:
Their sample app is far less involved than this. I would assume it's upstream bug (I wonder if anyone is actually using the plugin).

Reply:
I got the workaround here: https://github.com/rust-lang/cargo/issues/1706#issuecomment-563889929

Reply:
I think they already have a workaround: https://github.com/mozilla/rust-android-gradle/blob/3dc5511b2bbed4b66d03e360302b8147d7138877/plugin/src/main/kotlin/com/nishtahir/CargoBuildTask.kt#L193

Reply:
https://github.com/mozilla/rust-android-gradle/issues/30

Reply:
Yeah as I figured nobody is actually maintaining this plugin.